test/unit/mdc-toolbar/mdc-toolbar.test.js

```diff
@@ -0,0 +1,189 @@
+  const adjust = fixture.querySelector('.mdc-toolbar-fixed-adjust');
+  const component = new MDCToolbar(root);
+  return {root, adjust, component};
+}
+
+suite('MDCToolbar');
+
+test('attachTo initializes and returns an MDCToolbar instance', () => {
+  assert.isOk(MDCToolbar.attachTo(getFixture()) instanceof MDCToolbar);
+});
+
+test('adapter#hasClass returns true if the root element has specified class', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+  assert.isOk(component.getDefaultFoundation().adapter_.hasClass('foo'));
+});
+
+test('adapter#hasClass returns false if the root element does not have specified class', () => {
+  const {component} = setupTest();
+  assert.isNotOk(component.getDefaultFoundation().adapter_.hasClass('foo'));
+});
+
+test('adapter#addClass adds a class to the root element', () => {
+  const {root, component} = setupTest();
+  component.getDefaultFoundation().adapter_.addClass('foo');
+  assert.isOk(root.classList.contains('foo'));
+});
+
+test('adapter#removeClass removes a class from the root element', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+  component.getDefaultFoundation().adapter_.removeClass('foo');
+  assert.isNotOk(root.classList.contains('foo'));
+});
+
+test('adapter#registerScrollHandler registers the handler for scroll', () => {
+  const {component} = setupTest();
+  const handler = td.func('scrollHandler');
+  component.getDefaultFoundation().adapter_.registerScrollHandler(handler);
+  domEvents.emit(window, 'scroll');
+  td.verify(handler(td.matchers.anything()));
+  window.removeEventListener('scroll', handler);
+});
+
+test('adapter#deregisterScrollHandler unlistens the handler for scroll', () => {
+  const {component} = setupTest();
+  const handler = td.func('scrollHandler');
+  window.addEventListener('scroll', handler);
+  component.getDefaultFoundation().adapter_.deregisterScrollHandler(handler);
+  domEvents.emit(window, 'scroll');
+  td.verify(handler(td.matchers.anything()), {times: 0});
+  // Just to be safe
+  window.removeEventListener('scroll', handler);
+});
+
+test('adapter#registerResizeHandler registers the handler for window resize', () => {
+  const {component} = setupTest();
+  const handler = td.func('resizeHandler');
+  component.getDefaultFoundation().adapter_.registerResizeHandler(handler);
+  domEvents.emit(window, 'resize');
+  td.verify(handler(td.matchers.anything()));
+  window.removeEventListener('resize', handler);
+});
+
+test('adapter#deregisterResizeHandler unlistens the handler for window resize', () => {
+  const {component} = setupTest();
+  const handler = td.func('resizeHandler');
+  window.addEventListener('resize', handler);
+  component.getDefaultFoundation().adapter_.deregisterResizeHandler(handler);
+  domEvents.emit(window, 'resize');
+  td.verify(handler(td.matchers.anything()), {times: 0});
+  // Just to be safe
+  window.removeEventListener('resize', handler);
```

Comment:
Great thinking, but we need to wrap this in a `try/catch` as if the the `td.verify` above throws the test will be aborted.